test/router_xor_filter_SUITE.erl

```diff
@@ -147,6 +165,641 @@ publish_xor_test(Config) ->
     meck:unload(blockchain_worker),
     ok.
 
+max_filters_devices_test(Config) ->
+    Chain = proplists:get_value(chain, Config),
+    OUI1 = proplists:get_value(oui, Config),
+    Tab = proplists:get_value(ets, Config),
+
+    Round1Devices = n_rand_devices(10),
+    Round2Devices = n_rand_devices(10) ++ Round1Devices,
+    Round3Devices = n_rand_devices(10) ++ Round2Devices,
+    Round4Devices = n_rand_devices(10) ++ Round3Devices,
+    Round5Devices = n_rand_devices(10) ++ Round4Devices,
+
+    %% Init worker without processing first filter automatically
+    application:set_env(router, router_xor_filter_worker, false),
+    erlang:whereis(router_xor_filter_worker) ! post_init,
+
+    %% Wait until xor filter worker started properly
+    ok = test_utils:wait_until(fun() ->
+        State = sys:get_state(router_xor_filter_worker),
+        State#state.chain =/= undefined andalso
+            State#state.oui =/= undefined
+    end),
+
+    %% ------------------------------------------------------------
+    lists:foreach(
+        fun(#{devices := Devices, block := ExpectedBlock, filter_count := ExpectedFilterNum}) ->
+            true = ets:insert(Tab, {devices, Devices}),
+            ok = router_xor_filter_worker:check_filters(),
+
+            %% should have pushed a new filter to the chain
+            ok = expect_block(ExpectedBlock, Chain),
+
+            Filters1 = get_filters(Chain, OUI1),
+            ?assertEqual(ExpectedFilterNum, erlang:length(Filters1))
+        end,
+        [
+            #{devices => Round1Devices, block => 3, filter_count => 2},
```

Comment:
I thought that it would send all txns in one block?

Reply:
That's a fantastic question. Our `submit_txn` meck has it creating a block right away. Is that a problem here?

Reply:
After testing on a box, filter updates go into their own txns.